examples/generate-csr.py

```diff
@@ -0,0 +1,76 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# This script generates a simple SAN CSR to be used with Let's Encrypt
+# CA. Mostly intended for "auth --csr" testing, but, since it's easily
+# auditable, feel free to adjust it and use it on your production web
+# server.
+import os
+import argparse
+from OpenSSL import crypto
+
+try:
+    from cryptography.hazmat.backends import default_backend
+    from cryptography.hazmat.primitives.asymmetric import rsa, ec
+    from cryptography.hazmat.primitives.serialization import Encoding, PrivateFormat, NoEncryption
+    cryptography=True
+except ImportError:
+    cryptography=False
+
+CWD = os.getcwd()
+
+
+def generate_parser():
+    one_parser = argparse.ArgumentParser(description='Generate a simple SAN CSR')
+    one_parser.add_argument('domain', nargs='+', help='A FQDN to add in the CSR.')
+    one_parser.add_argument('--key-path', default=os.path.join(CWD, 'key.pem'),
+                            help='Path for the generated key '
+                                 '(default is key.pem in current directory)')
+    one_parser.add_argument('--csr-path', default=os.path.join(CWD, 'csr.der'),
+                            help='Path for the generated certificate '
+                                 '(default is csr.der in current directory)')
+    one_parser.add_argument('--key-type', default='RSA',
+                            choices=['RSA', 'ECDSA'],
+                            help='Key type to use, ECDSA is supported only '
+                                 'if cryptography module is available')
+
+    return one_parser
+
+
+def main(domains, key_path, csr_path, key_type):
+    san = ', '.join(['DNS: {0}'.format(item) for item in domains])
+
+    if key_type == 'RSA':
+        key = crypto.PKey()
+        key.generate_key(crypto.TYPE_RSA, 2048)
+    elif key_type == 'ECDSA':
```

Comment:
There's no reason to add a bunch of code we're not using.

Reply:
It will be used, in the tests. Previously it was called using openssl binary, but this is likely to be a problem on Windows. So I took also the occasion here to improve the script.

Alternatively, this may be included directly in misc for the tests, but I though interesting to improve this script.

Reply:
Postponed to next PR.